report/3.Theory/Parser.tex

```diff
@@ -0,0 +1,112 @@
+\section{Parser}
+
+The second part of the compiler will parse the lexical tokens coming from the lexer. 
+This part of the compiler is called the parser. 
+The parser works by taking a sequence of lexical tokens, and generating a parse tree or transforming them into an abstract syntax tree. 
+
+\subsection{Output data types}
+
+The parser will output some data type, that represents the compiled program in a hierarchical structure. 
+The outputted data type should hold all the information or almost all the information needed to generate the code later. 
+Sometimes the data type will be accommodated by a symbol table in order to generate the desired code.
+
+\textbf{Parse tree}
+
+One of the possible data structures to be outputted from the parser is the parse tree explained in section \ref{CFGdescription}. 
+The parse tree should resemble the structure of the CFG and the concrete input to the parser. 
+The parse tree is easily generated, and often the actual output from the parser. 
```

Comment:
I think you're missing something here (potentially an is)

Reply:
"and often is the output", måske.